### PR TITLE
chore: Remove isMobile when its possible

### PR DESCRIPTION
**Pull request by aniamusial**
# Scope of work
Removed isMobile observer and replaced it with css classes for following components:
- SfContentPages
- SfHeader
- SfHero
- SfBanner
- SfConfirmOrder

and following pages:
- Home
- Error page

# Screenshots of visual changes
-

# Checklist

- [ ] No commented blocks of code left
- [ ] Run tests and docs
- [ ] Self code-reviewed
- [ ] Changes documented 

If applicable:

- [ ] I followed [composition rules](https://docs.storefrontui.io/?path=/story/introduction-contributing-guide-code-guidelines--page) for my component
- [ ] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue [hunk 1]
```diff
@@ -1,10 +1,30 @@
 <template>
-  <section
-    class="sf-banner"
-    :style="style"
-    v-on="isMobileView ? $listeners : {}"
-  >
+  <section class="sf-banner" :style="style" v-on="$listeners">
     <component :is="wrapper" class="sf-banner__wrapper" :link="link">
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfBanner/SfBanner.vue:3]: Without mobile observer we can have div here - it won't change on button or link

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue [hunk 2]
```diff
@@ -92,13 +108,7 @@ export default {
       default: () => ({}),
     },
   },
-  data() {
-    return {
-      isMobileView: false,
```
Comment (justyna-13) [inline packages/vue/src/components/molecules/SfBanner/SfBanner.vue:97]: if this is removed then this line is not needed `return !this.isMobileView ? "div" : this.link ? "SfLink" : "SfButton";` line 128

File: packages/vue/src/components/organisms/SfHero/_internal/SfHeroItem.vue [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <template>
   <li class="glide__slide sf-hero-item" :style="style" data-testid="hero-item">
-    <component :is="wrapper" class="sf-hero-item__wrapper" :link="link">
+    <div
+      class="sf-hero-item__wrapper sf-hero-item__wrapper--desktop"
+      :link="link"
```
Comment (justyna-13) [inline packages/vue/src/components/organisms/SfHero/_internal/SfHeroItem.vue:5]: I think we can remove this link on div :)

**Review by AdamPawlinski — COMMENTED** [review]
SfHeader looks strange after adding to CT:
Screenshot from 2022-03-07 14-07-23

Maybe because mobile observer is not removed from SfHeaderNavigation component.

**Comment by aniamusial** [comment]
> SfHeader looks strange after adding to CT: Screenshot from 2022-03-07 14-07-23
> 
> Maybe because mobile observer is not removed from SfHeaderNavigation component.

@AdamPawlinski removed it from SfHeaderNavigation, please retest

**Comment by aniamusial** [comment]
> Looks like Content Pages on CT doesn't receive the proper value of isMobile and could not move to main catalog of MyAccount: content_pages_CT

@AdamPawlinski good catch, restored mobile observer for this component, but we should remove it in the next generations

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 1 above)
Comment (aniamusial) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue:3]: @justyna-13 We have div for desktop, but for mobile it must be a link/button as the whole banner is clickable then.

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 2 above)
Comment (aniamusial) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue:97]: done :)

File: packages/vue/src/components/organisms/SfHero/_internal/SfHeroItem.vue (hunk 3 above)
Comment (aniamusial) [reply packages/vue/src/components/organisms/SfHero/_internal/SfHeroItem.vue]: done :)

File: packages/vue/src/components/molecules/SfBanner/SfBanner.vue (hunk 1 above)
Comment (justyna-13) [reply packages/vue/src/components/molecules/SfBanner/SfBanner.vue:3]: right :) changed condition looks good

**Pull request merged**